### DISCUSSION TITLE
I am using TypeScript with NextJs, trying to add multiple fonts using variable, but getting error.

### DISCUSSION THREAD

**Discussion by madebyosama** [vercel/next.js#56798] [Help]
### Summary

const googleSansText = localFont({
  src: [
    {
      path: '../assets/fonts/GoogleSansText-Regular.woff2',
      weight: '400',
      style: 'normal',
      variable: '--google-sans-text-400',
    },

    {
      path: '../assets/fonts/GoogleSansText-Bold.woff2',
      weight: '700',
      style: 'normal',
    },
  ],
});

const googleSans = localFont({
  src: [
    {
      path: '../assets/fonts/GoogleSans-Regular.woff2',
      weight: '400',
      style: 'normal',
      variable: '--google-sans-400',
    },

    {
      path: '../assets/fonts/GoogleSans-Bold.woff2',
      weight: '700',
      style: 'normal',
    },
  ],
});

### Additional information

```js
Type '{ path: string; weight: string; style: string; variable: string; }' is not assignable to type 'string | { path: string; weight?: string | undefined; style?: string | undefined; }'.
  Object literal may only specify known properties, and 'variable' does not exist in type '{ path: string; weight?: string | undefined; style?: string | undefined; }'
```

**Comment by mhdalaa93** [comment]
Each `localFont` must have one `variable` property and it should be beside `src` property (i.e. directly inside `localFont` object not in each font weight).

Most likely you are using **TailwindCSS**, if so, you want to have CSS class for each of them, follow **Next.js Official Docs** for using localFont [With Tailwind CSS](https://nextjs.org/docs/app/building-your-application/optimizing/fonts#with-tailwind-css).

Accordingly, your code must be something like this:
```
const googleSansText = localFont({
  variable: "--google-sans-text",
  src: [
    {
      path: "../assets/fonts/GoogleSansText-Regular.woff2",
      weight: "400",
      style: "normal",
    },
    {
      path: "../assets/fonts/GoogleSansText-Bold.woff2",
      weight: "700",
      style: "normal",
    },
  ],
});

const googleSans = localFont({
  variable: "--google-sans",
  src: [
    {
      path: "../assets/fonts/GoogleSans-Regular.woff2",
      weight: "400",
      style: "normal",
    },
    {
      path: "../assets/fonts/GoogleSans-Bold.woff2",
      weight: "700",
      style: "normal",
    },
  ],
});
```

Then to use a different `weight` in a component, just add the variable name that you declared in `tailwind.config.ts` (after following the link above) and add a utility class for the needed weight. for example:
```
<div className="font-googleSansText font-normal">
<div className="font-googleSans font-bold">
```
